tastypie/resources.py

```diff
@@ -131,18 +126,15 @@ def __new__(cls, name, bases, attrs):
         declared_fields = {}
 
         # Inherit any fields from parent(s).
-        try:
-            parents = [b for b in bases if issubclass(b, Resource)]
-            # Simulate the MRO.
-            parents.reverse()
+        parents = [b for b in bases if issubclass(b, Resource)]
+        # Simulate the MRO.
+        parents.reverse()
 
-            for p in parents:
-                parent_fields = getattr(p, 'base_fields', {})
+        for p in parents:
+            parent_fields = getattr(p, 'base_fields', {})
 
-                for field_name, field_object in parent_fields.items():
-                    attrs['base_fields'][field_name] = deepcopy(field_object)
-        except NameError:
```

Comment:
This code was part of the very first commit. This except block has zero code coverage, and I can't think of how a NameError would get raised here.